### ISSUE TITLE
Slick w/ vanilla JS

### ISSUE THREAD

**Issue by interactionist** [kenwheeler/slick#494]
Hi Ken.
Thanks for this awesome slider, I really like it. But this time I would really like to use it w/o jQuery. 
Do you think it is possible to port it to vanilla JS?

Cheers.

**Comment by kenwheeler** [comment]
Closing, as this will be 2.0

**Comment by thymikee** [comment]
@kenwheeler, any ETA of v2.0?

**Comment by kenwheeler** [comment]
Working on it. The first step to 2.0 is getting CI and tests running. I'm almost there.

**Comment by thymikee** [comment]
+1

**Comment by alexbet** [comment]
+1

**Comment by SamHH** [comment]
+1

**Comment by DJviolin** [comment]
+1

**Comment by linnett** [comment]
+1

**Comment by dz000** [comment]
Hey Ken, are there any news about the vanilla version?
best regards

**Comment by Quirksmode** [comment]
+1 Any news of progress :-)

**Comment by xFlatlinex** [comment]
+1 subscribing for news on this subject :D

**Comment by controversial** [comment]
Is this still planned for v2 @kenwheeler ? Someone said v2 didn't seem to be moving in this direction.

**Comment by kenwheeler** [comment]
when v2 drops, it will be vanilla js

**Comment by controversial** [comment]
Awesome @kenwheeler. Any progress?

**Comment by kenwheeler** [comment]
a little but not a ton. I've been super focused on react and react-native lately, and driving initiatives at scale with both at my full time job. I would love to get v2 out the door, but I just havent had the hours in the day lately.

**Comment by controversial** [comment]
Ok. I don’t use jQuery as a matter of stubbornness, but if there’s a small
way in which I could help I’d love to.

In the meantime, are there competing libraries without jQuery dependency?
Slick’s feature set is just so awesome…

**Comment by kenwheeler** [comment]
Why not use jquery? If I wasn't in react land, I'd still use it. A couple extra kb in exchange for battle tested browser API normalization is well worth it imo.

**Comment by controversial** [comment]
Yeah, but I’m lucky enough to be 14 and not writing code that needs to be
cross-browser compatible. Thanks, though.

**Comment by ShaggyDude** [comment]
Microsoft has stopped supporting IE9. So should developers. 
+1 for javascript.

**Comment by controversial** [comment]
Currently going with http://flickity.metafizzy.co, for anyone interested. It has everything I need, and it's vanilla JS. When Slick drops jQuery I'll come back, though 😀

**Comment by websirnik** [comment]
I'm very happy with http://idangero.us/swiper it's flexible with good APIs.

**Comment by SamHH** [comment]
I've had a good experience with Swiper also, though I'll probably be moving back to Slick come v2 with this news. Many of the websites I develop now only support modern browsers so I've no need for jQuery, and it's massive bloat as a dependency.

Echoing someone above, if there's anything that can be done to help progress this cause let me/us know!

**Comment by kenwheeler** [comment]
Don't get me wrong, I get it, I wish I never had to support any older browsers, but slick is used on big ass websites where supporting older browsers means not losing out on potential revenue for them. If someone wants to, they could really just convert all the jquery to regular js. You would make an IIFE and pass a selector and then use QSA for all the selections and addEventListener for the events. Then use a lightweight animation lib for tweening values. If any of you wanna do that, I'd be happy to provide it as a vanilla version. V2 isn't about that. I mean v2 will be vanilla, but it would be a fundamental rewrite as well. Think free scrolling, dropping clones, physics based animation.

**Comment by controversial** [comment]
That's awesome @kenwheeler. Not hating on Slick. I'll switch back as soon
as v2 is out (or even in beta). You've done great work here already, as
well.

**Comment by thasmo** [comment]
@kenwheeler Does it work with cash.js?

**Comment by darrenjacoby** [comment]
@SamHH @websirnik had a look at Swiper, but it's double the size of Slick.  So, using jQuery and Slick comes to the same thing?

**Comment by splurtcake** [comment]
Any updates on v2 @kenwheeler ?

**Comment by alittlebirdo** [comment]
This is a good slick substitute until Slick drops jQuery: http://meandmax.github.io/lory/. Written in ES6 and looks/acts very similar to slick too.

**Comment by Quirksmode** [comment]
We found this alternative, it's vanilla JS and has so much of Slicks
functionality, it's brilliant.

http://idangero.us/swiper/#.WG5cEraLSWY

**Comment by darrenjacoby** [comment]
Thanks @alittlebirdo will have a look.  @Quirksmode I've had a look at Swiper, but it's larger than Slick and jQuery combined.

**Comment by splurtcake** [comment]
@alittlebirdo @Quirksmode @darrenjacoby a friend recommended this. Decided to go for it with the current project I'm working and. Really liking it so far.

http://flickity.metafizzy.co/

**Comment by darrenjacoby** [comment]
@splurtcake Looks good, but it has got a comm license, whereas Lory is MIT.

**Comment by benjamincharity** [comment]
I have worked with Flickity quite a bit and can say it's a really solid library.. Lory just seems to be quite a bit smaller than the other libraries.

**Comment by TheAsia** [comment]
@kenwheeler Its nice to have an approx time for v2 release. I really want to use it with React and no jquery.
 Thanks 😊

**Comment by mike3run** [comment]
@TheAsia there's actually a react port
https://github.com/akiran/react-slick

**Comment by SamHH** [comment]
@mike3run That appears to include jQuery under the hood based upon a cursory look, so it doesn't really solve the issue of needless dependency bloat.

**Comment by SamHH** [comment]
@mike3run Ah I think you're right. Strange that jQuery is listed under Bower deps though.

**Comment by mike3run** [comment]
i mean bower got deprecated a while ago so most likely not all packages are updated in that department :(

**Comment by kalpeshmange** [comment]
@kenwheeler We're awaiting the v2 release ! Please let us shower in the holy rains of Vanilla JS Slick! :)

**Comment by mike3run** [comment]
@kalpeshmange give [tiny-slider](https://github.com/ganlanyuan/tiny-slider) a try, its vanilla, works really well and the developer is actively adding new stuff and ironing bugs

**Comment by rudza** [comment]
3 years since this issue was opened. :( 
+1 for vanilla

**Comment by mike3run** [comment]
@rudza you know @kenwheeler doesn't owe us anything right?

if you want the vanilla so badly make a PR, or give tiny-slider a try, but don't just straight up demand features like you're paying for them 😛

**Comment by kucharzgotuje** [comment]
Great slider, love your work, but I have also dreamed about Vanilla Slick, just saying.
++
Regards

**Comment by mike3run** [comment]
Then build it yourself or back someone like tiny slider with money so they
can focus on fulfilling your dreams better

**Comment by akroticssumit** [comment]
+1

**Comment by xttn** [comment]
+1

**Comment by pthurmond** [comment]
I will be honest, it is a bit of a dick move to constantly add references to these other competing projects in these tickets. I am not even a developer on this project, but as a developer in general this would really piss me off. It is insulting to the developers that have worked so hard to build this great tool that a ton of websites use.

Yeah the jQuery limitation is a blocker for some people. But I am sure these guys have other things to do then just code to everyone's requests. If you want that took then fork it and PR back to this project with your VanillaJS version of this tool. If you build it and do so very well then I am sure they will strongly consider your changes.

If you are on this forum then you probably are developer. If you are a developer then you are capable of doing this refactor yourself. Give back to the community and do this refactor.

**Comment by kucharzgotuje** [comment]
Yeah, I understand your point but look on the other side of the issue:
> slick is the best
> jquery is great
> slick is the best but jquery gets crappy
> you have tested algoryrthms & you're pretty experienced
> you are very popular at this point, decide do move towards faster vanilla
js
> you have better product, you're populating the market gap
> slick is still the best
> profit.

jQuery will die eventually. Many people are moving towards pure js / es6
transpiled. People are creating similar sliders becouse there's a need for
such. (I would test them but, you know, slick for the win) MaterializeCss
has done that as well, they moved all theirs library to vanilla js.

I know they have better things to do, I understand it's annoying, but
there's a need, simple.





<https://mailtrack.io/> Sent with Mailtrack
<https://chrome.google.com/webstore/detail/mailtrack-for-gmail-inbox/ndnaehgpjlnokgebbaldlmgkapkpjkkb?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality>

2018-02-13 19:55 GMT+01:00 Patrick Thurmond <notifications@github.com>:

**Comment by kylegoines** [comment]
just checking up on this, still a thing asked most for at my studio :)

**Comment by DJviolin** [comment]
I like slick slider a lot, but we have to ask ourselves: what we gain from a vanilla rewrite? Performance? Less filesize? More stability (or the opposite)? We are in 2018 and we live in the age of client side frameworks. Just take a look what is possible with vue.js:

https://vuejsexamples.com/tag/slider/

Of course, there are a lot of plugins in here as well, like the slick slider addon for vue:

https://vuejsexamples.com/vue-component-for-slick-carousel/

Lot of these projects are just pure vue components making your life easier that you don't have to write them yourself:

https://github.com/SSENSE/vue-carousel

Lot of people complaining about ditching Jquery, because we don't use it anymore. Not realizing that they asking something for a bigger final project size, if they use a client side framework anyway.

**Comment by shafqatdotnet** [comment]
Use this one without any dependencies & pure js : https://github.com/pawelgrzybek/siema

**Comment by notrealdev** [comment]
4 years have passed, I think we should forget this issue

**Comment by mattmattvanvoorst** [comment]
+1

**Comment by gonzarodriguezt** [comment]
Well i'm gonna try to learn enough and someday make the refactor :)

**Comment by eurolife** [comment]
+1 
On most projects I work on, I prefer to use JQuery so all is good, but on a current project, they want to dump JQuery to save in total JS file size so I'm forced to find another slider that doesn't require it. Shame, as I love this one, it's perfect.

**Comment by darrenjacoby** [comment]
[tinyslider](https://github.com/ganlanyuan/tiny-slider) is pretty good.

**Comment by pablo67340** [comment]
> 4 years...I think we should forget this issue

Never quit!

**Comment by eurolife** [comment]
I ended up making my own one, but this one is still better.

**Comment by darrenjacoby** [comment]
Has anyone tried using [cash.js](http://kenwheeler.github.io/cash/) and this? Could be a good alternative to pulling in jQuery and still using this.

**Comment by pablo67340** [comment]
> Has anyone tried using [cash.js](https://github.com/kenwheeler/cash/) and Slick? Could be a good alternative to pulling in jQuery and still using this.

Would be very cool to see imo. I personally avoid using JQuery on all of my websites period.

**Comment by pm-designs** [comment]
Man .... i thought this would be the the last carousel I willl ever need :P
Slick Slider is an awesome slider, hope you still working on this :)

**Comment by SephReed** [comment]
If you're looking for something simple and no-modules-vanilla, this is the best I've found: https://nickpiscitelli.github.io/Glider.js/

It's missing a lot of the features, but it's should cover at least ~75% of use cases.

**Comment by ico85** [comment]
> [tinyslider](https://github.com/ganlanyuan/tiny-slider) is pretty good.

Very buggy. A lot of issues on Github.

**Comment by OktarinTentakel** [comment]
> Has anyone tried using [cash.js](https://github.com/kenwheeler/cash/) and Slick? Could be a good alternative to pulling in jQuery and still using this.

Yes, I actually did today and got it running with a few minor additions to cash and some smallish tweaks to Slick 1.8.1.

I did not test in IE11, but everything Edge 18 and up seems to be running generally pretty smoothly.

I will test this further, because this is actually for a real project, but the results seem promising so far. I'm quite positive that this will hold up with different configs, since I chose a rather complex one from the get go. Still have to verify if instance handling with multiple sliders on one page holds up and if destroying and reinitialization works properly.

Just give me a mention or PM if you are interested in the changes.

**Comment by CorySchulz** [comment]
> > Has anyone tried using [cash.js](https://github.com/kenwheeler/cash/) and Slick? Could be a good alternative to pulling in jQuery and still using this.
> 
> Yes, I actually did today and got it running with a few minor additions to cash and some smallish tweaks to Slick 1.8.1.
> 
> I did not test in IE11, but everything Edge 18 and up seems to be running generally pretty smoothly.
> 
> I will test this further, because this is actually for a real project, but the results seem promising so far. I'm quite positive that this will hold up with different configs, since I chose a rather complex one from the get go. Still have to verify if instance handling with multiple sliders on one page holds up and if destroying and reinitialization works properly.
> 
> Just give me a mention or PM if you are interested in the changes.

I was just about to try the same thing. Keep us posted if it's easy to use cash.js in place of jquery.

**Comment by OktarinTentakel** [comment]
@CorySchulz 

I created a gist for you, outlining our current solution.

https://gist.github.com/OktarinTentakel/6480bdd1b0277b2d75afab8bb2b8d603

We had to fill some gaps in cash to work with slick (first file "cash-shim.js ", methods starting with underscore are from lodash, mainly for type checks).

Then we took version 1.8.1 of slick slider and mainly reworked some details of event handling (second file "slick.mod.js "). All changed lines should be marked with "// MOD". Additionally we removed the boilerplate module stuff and just encapsulated into a factory method that we can import more easily in ES6 (you can see the call in the first file).

Have fun!

**Comment by chinhphan099** [comment]
Any update Slick slider vanila Js version? @kenwheeler

**Comment by CorySchulz** [comment]
@OktarinTentakel 

I was able to get this to work! I actually spent the day modding the cash and slick js files and essentially recoded the same things you did, but I'm going to use your shim file because I think that's a cleaner implementation and modifies the original Slick file the least. You should package this and post it online and call it "Slick Cash". I'm sure a ton of people would use this. 
Thanks again!

**Comment by kylegoines** [comment]
> @OktarinTentakel

please do, i dont have much stake in this as ive moved to other solutions, but for me slick is dear to my heart

**Comment by chinhphan099** [comment]
Current I'm using swiperjs for vanilla js project. It is used in conjunction with Flebox. That's good.
But slick slider went with me through a long time from I became a coder.
So I'm expecting slick slider vanilla version :)

**Comment by OktarinTentakel** [comment]
I honestly tried most go-to sliders for a new project and was quite disappointed with the offerings. Slick still offers some layout options no other slider is able to pull off, which out creative guys really seem to like.

I tried all three current big vanilla solutions and none was satisfactory. That's when I tried the port.

I'll see if I can find the time to put the changes into a neater package somehow.

**Comment by ubergeoff** [comment]
Busy working on a vanilla port...

Got quite far (still some work to do) - could use some help - if anyone is willing to contribute:

https://github.com/ubergeoff/flingo/blob/master/apps/flingo/src/vendor/slick/slick.js

**Comment by sheriffmarley** [comment]
@ubergeoff I just found your vanillajs version of slick and I'm curious what's the state of affairs? Were you able to make a fully independent jquery version?

**Comment by ubergeoff** [comment]
HI there @sheriffmarley 

Sorry - got a little off track for a while - but I am back on it now.

However: I have made good progress since Aug 26.

As for a status update:

1) I have remove Jquery animations - as CSS transitions options are just fine.
2) I am still trying to cover all the events - it's been a little challenging (converting these away from JQuery) - but I feel I have made good progress.
3) After 2) is completed - I believe I will have a final script that is good to go.

DEMO's can be seen here: https://www.flingo.co.za/slick/slick1

A few more days are going to be required for me to finish off a add some polish. (Please hold thumbs)

As always: any code contributions are welcome...

**Comment by ubergeoff** [comment]
Ok job 99% done. But mostly useable.

Still outstanding (WIP)

1) Accessibility features
2) Infinite scrolling

Full source: https://github.com/ubergeoff/flingo
And if I made your day: https://www.flingo.co.za/donate

**Comment by chinhphan099** [comment]
Hi @ubergeoff ,

Could you please share the url of slick slider in vanilla JS files?

**Comment by ubergeoff** [comment]
Hi there @chinhphan099 

Not 100% sure what you are requesting..?

Are you asking slick script (using vanilla js)? If yes - then here it is: https://github.com/ubergeoff/flingo/blob/master/apps/flingo/src/vendor/slick/slick.js

Or are you asking for something else..?

**Comment by chinhphan099** [comment]
Thanks @ubergeoff .
Much appreciate. I'm going to apply this file https://github.com/ubergeoff/flingo/blob/master/apps/flingo/src/vendor/slick/slick.js into my project.

**Comment by ubergeoff** [comment]
> Hi @ubergeoff ,
> 
> Could you please share the url of slick slider in vanilla JS files?



> Thanks @ubergeoff .
> Much appreciate. I'm going to apply this file https://github.com/ubergeoff/flingo/blob/master/apps/flingo/src/vendor/slick/slick.js into my project.

OK @chinhphan099 - please let me know how it goes...

**Comment by chinhphan099** [comment]
I tried and face a issue: https://i.gyazo.com/6559ab687bc8011f91ff15743b968360.mp4
Current I stopped debug that issue.

So I'm using the slick with jquery to push live my project.

I am expecting your final vanilla js version to use, @ubergeoff.

It would be great if you make a web page to explain the difference between your version with Kenwheeler (if have)

**Comment by ubergeoff** [comment]
> So I'm using the slick with jquery to push live my project.
> 
> I am expecting your final vanilla js version to use, @ubergeoff.
> 
> It would be great if you make a web page to explain the difference between your version with Kenwheeler (if have)

I think, lets take this over to another thread...

PLease can we try resolve this issue here: https://github.com/ubergeoff/flingo/issues/1

**Comment by grimcap** [comment]
https://flickity.metafizzy.co/

**Comment by varave** [comment]
2021. the issue is still not resolved 😨

**Comment by ubergeoff** [comment]
HI there @varave

What is not resolve for you..?

Are you not 100% happy with: https://github.com/kenwheeler/slick/issues/494#issuecomment-704821512 

?

**Comment by varave** [comment]
Hi @ubergeof!
Now i use [vue-agile](https://github.com/lukaszflorczak/vue-agile) in my vue.js project

**Comment by FacundoGamond** [comment]
@kenwheeler this is the best slider in the world I know, its a shame that still use JQuery, Why don't just use patreon to buy you a coffie?

**Comment by gtempesta-pixartprinting** [comment]
Hi @ubergeoff

I'm trying to progressively remove jQuery from a web app, so using your version of the library seems like a good start, bu I'm having some issues, as it looks like jQuery is not completely removed.

For example I had to change this line
```js
_.$slideTrack
  .find('.slick-cloned')
  .find('[id]')
  .each(function () {
    $(this).attr('id', '');
  });
```
to this
```js
$(_.$slideTrack)
  .find('.slick-cloned')
  .find('[id]')
  .each(function () {
    $(this).attr('id', '');
});
```
in order to prevent an error.

Now I'm trying to destroy and create the slideshow but if I run 
```js
const uspSlideshow = new Slick(uspSlide, uspOptions);
slickSlider.unslick();
slickSlider.reinit();
```
I get an error which I'm not able to fix:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'offsetWidth')
```

Do you think I can open an issue on the https://github.com/ubergeoff/flingo/ repository even if it's only related to a library in that project?

Thanks

**Comment by ubergeoff** [comment]
@gtempesta-pixartprinting - please feel free to log the error there - yes... will take a look over on that side..

**Comment by umkasanki** [comment]
+1

**Comment by AGrush** [comment]
So how do you use this in a basic one page website?

**Comment by ronfarbernewman** [comment]
@ubergeoff Thank you so much for your work on vanilla-fy-ing my favorite slider plugin!

@AGrush (and anyone finding this thread in the future), I had luck setting up [ubergeoff's version of Slick](https://github.com/ubergeoff/flingo/blob/master/apps/flingo/src/vendor/slick/slick.js) like this for usage outside a framework.

I needed to reference my Slicks from a separate JS file than they were created in, so I decided to store them on the `window` object, referencing them by the name/id of the parent. 

```
var containerID = "my-container-id";
var slickContainer = document.getElementById(containerID);
var slickObject = new Slick(slickContainer, {
  slidesToScroll: 1,
  slidesToShow: 3,
  arrows: false,
  dots: false,
  mobileFirst: true
});

var Slicks = window.Slicks = window.Slicks || {};
Slicks[containerID] = slickObject;
```

With it setup like that, elsewhere after my Slick has been initialized, I can call the Slick methods on the Slick item, like this:

```
window.Slicks["my-container-id"].slickGoTo(slideNumber, true);
```

Hope that helps anyone looking to do the same thing.
